src/mobster/tekton/common.py

```diff
@@ -120,24 +126,96 @@ def upload_to_atlas(dirpath: Path, atlas_url: str) -> None:
         raise AtlasUploadError() from err
 
 
-async def upload_to_s3(dirpath: Path, bucket: str) -> None:
+def connect_with_s3(retry_s3_bucket: str) -> S3Client | None:
     """
-    Upload SBOMs to S3 bucket.
+    Connect with AWS S3 using S3Client.
 
     Args:
-        dirpath: Directory containing SBOMs to upload.
-        bucket: S3 bucket name.
+        retry_s3_bucket: S3 bucket name.
+
+    Returns:
+        client: S3Client object
     """
+    if not retry_s3_bucket:
+        return None
+    if not s3_credentials_exist():
+        raise ValueError("Missing AWS authentication.")
     client = S3Client(
-        bucket=bucket,
+        bucket=retry_s3_bucket,
         access_key=os.environ["AWS_ACCESS_KEY_ID"],
         secret_key=os.environ["AWS_SECRET_ACCESS_KEY"],
         endpoint_url=os.environ.get(
             "AWS_ENDPOINT_URL"
         ),  # configurable for testing purposes
     )
 
-    await client.upload_dir(dirpath)
+    return client
+
+
+async def upload_to_s3(s3_client: S3Client | None, dirpath: Path) -> None:
+    """
+    Upload SBOMs to S3 bucket.
+
+    Args:
+        s3_client: S3Client object
+        dirpath: Directory containing SBOMs to upload.
+    """
+    if s3_client is None:
+        return
+    await s3_client.upload_dir(dirpath)
+
+
+def validate_sbom_input_data(
+    sbom_input_file: Path,
+    obj: type[SnapshotModel] | type[ReleaseData],
+) -> Any:
+    """
+    Store SBOM Input data for release_data to S3 bucket.
```

Comment:
This docstring seems to be wrong to me. The function loads SBOM data, validates it, and returns it.